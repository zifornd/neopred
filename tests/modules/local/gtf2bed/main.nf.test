nextflow_process{
	name "Test Process GTF2BED"
    script "modules/local/gtf2bed/main.nf"
    process "GTF2BED"
    //config "${projectDir}/tests/nextflow.config"
    
    tag "modules"
    tag "modules_local"
    tag "gtf2bed"

    test("Should run GTF2BED without failures") {

        when {
            process {
                """
                // Print the file path for debugging
                println "GTF file: " + params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gtf'
                input[0] = [
                   // [id: 'test_gtf'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gtf', checkIfExists: true)
                ]
                
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.bed}, //.bed[0] ==~ ".*/genome.bed" },
                { process.out.versions}
            )
        }

    }

    test("Should run GTF2BED with stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                   // [id: 'test_gtf'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gtf', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.bed},
                { process.out.versions}
            )
        }

    }

}
