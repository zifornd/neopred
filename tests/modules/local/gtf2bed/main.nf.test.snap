{
    "Should run GTF2BED with stub": {
        "content": [
            {
                "0": [
                    "genome.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,3edee107ec8242f5d3f4a89db7a1a8d5"
                ],
                "bed": [
                    "genome.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,3edee107ec8242f5d3f4a89db7a1a8d5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-07T07:03:11.268616"
    }
}